custom_components/jablotron_cloud/config_flow.py

```diff
@@ -1,73 +1,131 @@
 """Config flow for Jablotron Cloud integration."""
-from __future__ import annotations
 
 import logging
-from typing import Any
 
-from jablotronpy import Jablotron, UnexpectedResponse
 import voluptuous as vol
-
 from homeassistant import config_entries
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import HomeAssistant
-from homeassistant.data_entry_flow import FlowResult
-from homeassistant.exceptions import HomeAssistantError
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_PIN, CONF_FORCE_UPDATE, CONF_SCAN_INTERVAL, \
+    CONF_TIMEOUT
+from jablotronpy import UnauthorizedException
 
 from .const import DOMAIN
+from .jablotron import JablotronClient
 
 _LOGGER = logging.getLogger(__name__)
 
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-    }
-)
 
+def get_schema(config: dict) -> vol.Schema:
+    """Return config flow schema."""
+
+    return vol.Schema(
+        {
+            vol.Required(CONF_USERNAME, default=config.get(CONF_USERNAME, "")): str,
+            vol.Required(CONF_PASSWORD): str,
+            vol.Optional(CONF_PIN, default=config.get(CONF_PIN, "")): str,
+            vol.Optional(CONF_FORCE_UPDATE, default=config.get(CONF_FORCE_UPDATE, True)): bool,
+            vol.Optional(CONF_SCAN_INTERVAL, default=config.get(CONF_SCAN_INTERVAL, 30)): int,
```

Comment:
I'm not a fan on allowing user to change scan interval.
1) Minimum is 30 and users will try to add lower values. This should not be allowed. We might overload the API and get banned.
2) Why would you need higher interval? I need lower but higher does nto make any sense.

Reply:
I think it should be the user's concern what they set. 
Originally it was in the README section of the integration limits. I added it as an option for the user to customize the settings. Maybe someone will be fine with a higher limit, maybe someone will want 20 seconds.

Either way, the 30 second interval is used by default so the user doesn't have to change anything.